src/rolling/directory.rs

```diff
@@ -184,13 +189,15 @@ impl BlockRead for RollingReader {
             self.block_id += 1;
             return Ok(true);
         }
+
+        let mut next_file_number =
+            if let Some(next_file_number) = self.directory.files.next(&self.file_number) {
+                next_file_number
+            } else {
+                return Ok(false);
+            };
+
         loop {
-            let next_file_number =
```

Comment:
I don't understand the difference here?

Reply:
it's related to the new test `test_read_truncated`: assuming a scenario where we manage to create a file, but get terminated  before resizing it (or it getting truncated to less than a block worth of data, for any reason). The previous version would get into an infinite loop over that file, whereas the new version guarantee progress.
An alternative change would have been to move `self.file_number = next_file_number` out of the if block just below, but I wasn't sure if it would break a non obvious invariant somewhere else